README.md

```diff
@@ -41,6 +41,32 @@ Some things cannot be preloaded, use `predictive_load: false`
 has_many :foos, predictive_load: false
 ```
 
+### Instrumentation
+
+The library can be instrumented by providing a callback, to be invoked every time automatic preloading happens. The callback must be a callable that receives two arguments:
+* The record (instance) on which the queries that triggered automatic preloading are being performed, in the form of some association call.
+* The association object, which can be inspected to check the type and name of the association.
+
+For example, the callback could be used to emit some metrics:
+
+```ruby
+require "active_support/core_ext/string"
+
+PredictiveLoad.config do |c|
+  c.callback = -> (record, association) do
+    return unless instrument_preloads?
+
```

Comment:
method does not exist, so maybe don't mention that
```suggestion
```

Reply:
Done.